gui/src/components/WithLeftWorkspaceNav.svelte

```diff
@@ -0,0 +1,102 @@
+<script lang="ts">
+  import * as router from 'svelte-spa-router';
+
+  import Home from './mono/home.svelte';
+  import Layers from './mono/layers.svelte';
+  import Storage from './mono/database.svelte';
+  import Networking from './mono/networking.svelte';
+
+  export let workspaceId: string;
+  export let active: string;
+</script>
+
+<div>
+  <aside>
+    <button
+      on:click={() => {
+        router.push(`#/workspaces/${workspaceId}`);
+      }}
+      class={active === 'Dashboard' ? 'active' : ''}
+      title="Dashboard"><Home /></button
+    >
+    <button
+      on:click={() => {
+        router.push(`#/workspaces/${workspaceId}/components`);
+      }}
+      class={active === 'Components' ? 'active' : ''}
+      title="Components"><Layers /></button
+    >
+    <button
+      on:click={() => {
+        router.push(`#/workspaces/${workspaceId}/storage`);
+      }}
+      class={active === 'Storage' ? 'active' : ''}
+      title="Storage"><Storage /></button
+    >
+    <button
+      on:click={() => {
+        router.push(`#/workspaces/${workspaceId}/networking`);
+      }}
+      class={active === 'Networking' ? 'active' : ''}
+      title="Networking"><Networking /></button
+    >
+  </aside>
+  <section>
+    <slot />
```

Comment:
As mentioned elsewhere, I think that this component should _only_ be the sidebar and let the Layout component determine where to use the sidebar and main content.